spec/parser_spec.rb

```diff
@@ -85,17 +85,26 @@
   
   if RUBY_VERSION > "1.9.0"
     it "should preserve character encoding" do
-      input = "This is an ISO-8859-1 string"
-      input.force_encoding 'iso-8859-1'
+      if RUBY_VERSION > "2.3.0"
+        input = String.new("This is an ISO-8859-1 string", encoding: 'iso-8859-1')
+      else
+        input = "This is an ISO-8859-1 string"
+        input.force_encoding 'iso-8859-1'
+      end
+
       output = RedCloth.new(input).to_html
       
       output.should == "<p>This is an <span class=\"caps\">ISO</span>-8859-1 string</p>"
       output.encoding.to_s.should == "ISO-8859-1"
     end
     
     it "should not raise ArgumentError: invalid byte sequence" do
-      s = "\xa3"
-      s.force_encoding 'iso-8859-1'
+      if RUBY_VERSION > "2.3.0"
```

Comment:
Rather than having version-specific logic here, I'd recommend just making the initial string mutable: `s = "\xa3".dup`